packages/hub/src/lib/hardware-requirements.ts

```diff
@@ -0,0 +1,72 @@
+import { ListFileEntry, listFiles } from "./list-files";
+
+export interface MemoryRequirements {
+	minimumGigabytes: number;
+	recommendedGigabytes: number;
+};
+
+export interface HardwareRequirements {
+	name: string;
+	memory: MemoryRequirements;
+};
+
+export async function getHardwareRequirements(params: {
+	/**
+	 * The model name in the format of `namespace/repo`.
+	 */
+	name: string;
+	/**
+	 * The context size in tokens, default to 2048.
+	 */
+	contextSize?: number;
```

Comment:
if possible `contextSize` should be taken in `tasks.ModelData` so it's clearer the data is coming from normalized parsing of HF model repos

Reply:
and tbh i'm wondering if this kinda of method should better live in `tasks` or `gguf` modules rather than here

Reply:
IMO the `contextSize` should be a number that most users gonna use in practice, for example 2048 or 4096

Setting it to the max `contextSize` of the model may not be useful, as most users never have enough VRAM to run full context length anyway (especially, 128k context are more and more common nowadays)

Reply:
ok, i see your point. cc @gary149 wdyt?
maybe it's another selector in the UI then with "user-set context size"?